dozer-tests/src/sqllogictest/suits/join.test

```diff
@@ -0,0 +1,308 @@
+INSERT INTO film_actor (actor_id, film_id, last_update) VALUES (2, 6, '2022-03-16 10:00:00');
+
+statement ok
+INSERT INTO film_actor (actor_id, film_id, last_update) VALUES (3, 7, '2022-03-16 10:00:00');
+
+statement ok
+INSERT INTO film_actor (actor_id, film_id, last_update) VALUES (3, 8, '2022-03-16 10:00:00');
+
+statement ok
+INSERT INTO film_actor (actor_id, film_id, last_update) VALUES (3, 9, '2022-03-16 10:00:00');
+
+statement ok
+INSERT INTO film_actor (actor_id, film_id, last_update) VALUES (4, 10, '2022-03-16 10:00:00');
+
+
+query ITT
+SELECT actor.actor_id, first_name, last_name from actor
+        JOIN film_actor on film_actor.actor_id = actor.actor_id
+----
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+2 jack nicholson
+2 jack nicholson
+2 jack nicholson
+3 angelina jolie
+3 angelina jolie
+3 angelina jolie
+4 tom hanks
+
+statement ok
+INSERT INTO actor(actor_id,first_name, last_name, last_update) values (2, 'dario', 'guiness','2020-02-15 09:34:33+00')
+
+statement ok
+INSERT INTO actor(actor_id,first_name, last_name, last_update) values (3, 'luigi', 'guiness','2020-02-15 09:34:33+00')
+
+query ITT
+SELECT actor.actor_id, first_name, last_name from actor
+        JOIN film_actor on film_actor.actor_id = actor.actor_id
+----
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+2 dario guiness
+2 dario guiness
+2 dario guiness
+2 jack nicholson
+2 jack nicholson
+2 jack nicholson
+3 angelina jolie
+3 angelina jolie
+3 angelina jolie
+3 luigi guiness
+3 luigi guiness
+3 luigi guiness
+4 tom hanks
+
+query I
+SELECT a.actor_id from actor a
+        JOIN film_actor fa on fa.actor_id = a.actor_id
+        JOIN film f on f.film_id = fa.film_id
+----
+1
+1
+1
+1
+1
+1
+2
+2
+2
+2
+2
+2
+3
+3
+3
+3
+3
+3
+4
+
+query ITT
+SELECT a.actor_id, a.first_name, a.last_name from actor a
+        JOIN film_actor fa on fa.actor_id = a.actor_id;
+----
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+2 dario guiness
+2 dario guiness
+2 dario guiness
+2 jack nicholson
+2 jack nicholson
+2 jack nicholson
+3 angelina jolie
+3 angelina jolie
+3 angelina jolie
+3 luigi guiness
+3 luigi guiness
+3 luigi guiness
+4 tom hanks
+
+
+query ITT
+WITH tbl AS (SELECT actor_id, first_name, last_name FROM actor)
+        SELECT t.actor_id, t.first_name, t.last_name
+        FROM tbl t
+        JOIN film_actor fa ON fa.actor_id = t.actor_id;
+----
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+1 penelope guiness
+2 dario guiness
+2 dario guiness
+2 dario guiness
+2 jack nicholson
+2 jack nicholson
+2 jack nicholson
+3 angelina jolie
+3 angelina jolie
+3 angelina jolie
+3 luigi guiness
+3 luigi guiness
+3 luigi guiness
+4 tom hanks
+
+# dozer has bug for the sql, less one row `1 penelope guiness`
```

Comment:
I find a left join bug  @mediuminvader

Reply:
Nice, but I'm refactoring the join right now #1180 , let's re run this on the next merge